components/org.wso2.carbon.identity.application.authenticator.push.core/src/main/java/org/wso2/carbon/identity/application/authenticator/push/core/PushJWTValidator.java

```diff
@@ -0,0 +1,221 @@
+     */
+    public boolean validate(String jwt, String publicKey)
+            throws IdentityPushException {
+
+        if (!isJWT(jwt)) {
+            return false;
+        }
+
+        try {
+            SignedJWT signedJWT = SignedJWT.parse(jwt);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            if (claimsSet == null) {
+                throw new IdentityPushException("Claim values are empty in the given jwt.");
+            }
+
+            // JWT Signature validation
+            if (!validateSignature(publicKey, signedJWT)) {
+                return false;
+            }
+
+            // Validation of expiry time
+            if (!checkExpirationTime(claimsSet.getExpirationTime())) {
+                return false;
+            }
+            // Validation of active time
+            if (!checkNotBeforeTime(claimsSet.getNotBeforeTime())) {
+                return false;
+            }
+        } catch (ParseException e) {
+            throw new IdentityPushException("Error while validating jwt", e);
+        }
+        return true;
+    }
+
+    /**
+     * Validate the signature of the JWT.
+     *
+     * @param publicKeyStr Public key for used for signing the JWT
+     * @param signedJWT    Signed JWT
+     * @return Boolean value for signature validation
+     * @throws IdentityPushException Error when validating the signature
+     */
+    public static boolean validateSignature(String publicKeyStr, SignedJWT signedJWT) throws IdentityPushException {
+
+        try {
+            byte[] publicKeyData = Base64.getDecoder().decode(publicKeyStr);
+            X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+            KeyFactory kf = KeyFactory.getInstance("RSA");
+            RSAPublicKey publicKey = (RSAPublicKey) kf.generatePublic(spec);
+
+            JWSVerifier verifier = new RSASSAVerifier(publicKey);
+            return signedJWT.verify(verifier);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | JOSEException e) {
+            throw new IdentityPushException("Error occurred when validating token signature.", e);
+        }
+    }
+
+    /**
+     * Get claim values from JWT by name.
+     *
+     * @param jwt   JWT string
+     * @param claim Name of the required claim
+     * @return Retrieved claim
+     */
+    public String getJWTClaim(String jwt, String claim) {
+
+        try {
+            if (!isJWT(jwt)) {
+                return null;
+            } else {
+                SignedJWT signedJWT = SignedJWT.parse(jwt);
+                JWTClaimsSet claimSet = signedJWT.getJWTClaimsSet();
+                if (claimSet != null) {
+                    return (String) claimSet.getClaim(claim);
+                } else {
+                    return null;
+                }
+            }
+        } catch (ParseException e) {
+            if (!log.isDebugEnabled()) {
+                log.error("Error occurred while parsing JWT string", e);
```

Comment:
Throw exception

Reply:
Returns true or false as prior. An exception is thrown at receiving end if result is false